UI presigned URL (#5239)
treeverse/lakeFS: docs/reference/presigned-url.md

---
layout: default
title: Presigned URL
description: Configuring lakeFS to use presigned URLs.
parent: Reference
nav_order: 100
has_children: false
---

# Presigned URL
{: .no_toc }

{% include toc.html %}

> By default, lakeFS will get and set objects using its server. lakeFS also support accessing objects using a presinged URL which allows direct access to the object.

Author: done

> lakeFS assume by default your object storage support presinged URLs, if not it should be disabled in the configuration.

Reviewer: Do you mean the UI?

Author: No, if the object storage doesn't support presigned URLs in general, like MinIO

# Using presigned URLs in the UI

> For lakeFS to fetch objects through the UI using a presigned URL (instead of getting it with lakeFS) need to enable the presigned URL support UI in the lakeFS configuration and add cors permissions to the bucket.

Reviewer: 1. Think we need to explain that the UI fetch the objects - its not lakeFS. The lakeFS is the source that the UI can fetch objects from.
2. cors -> CORS (Cross-Origin Resource Sharing) + suggest reference the formal definition

Author: There is link on each object store to enable CORS where it is explained

> **⚠️ Note** currently duckDB do not support presigned URL.

Reviewer: ```suggestion
**⚠️ Note** Currently DuckDB fetching data from lakeFS does not support fetching data using presigned URL.
```

Author: done

### Example: [AWS S3](https://docs.aws.amazon.com/AmazonS3/latest/userguide/enabling-cors-examples.html)

   ```json

>   [

Reviewer: not aligned

Author: done

    {
        "AllowedHeaders": [
            "*"
        ],
        "AllowedMethods": [
            "GET",
            "PUT"
        ],
        "AllowedOrigins": [
            "lakefs.endpoint"
        ],
        "ExposeHeaders": [
            "ETag"
        ]
    }
  ]
   ```

### Example: [Google Storage](https://cloud.google.com/storage/docs/using-cors)

   ```json

>   [

Author: done

    {
        "origin": ["lakefs.endpoint"],
        "responseHeader": [
            "ETag"],
        "method": ["PUT", "GET"],
        "maxAgeSeconds": 3600
    }
   ]
  ```

### Example: [Azure blob storage](https://learn.microsoft.com/en-us/rest/api/storageservices/cross-origin-resource-sharing--cors--support-for-the-azure-storage-services)

```xml
  <Cors>
      <CorsRule>  
          <AllowedOrigins>lakefs.endpoint</AllowedOrigins>  
          <AllowedMethods>PUT,GET</AllowedMethods>  
          <AllowedHeaders>*</AllowedHeaders>  
          <ExposedHeaders>ETag</ExposedHeaders>  
          <MaxAgeInSeconds>3600</MaxAgeInSeconds>  
      </CorsRule>  
  </Cors>

>   ```

Author: done

---
layout: default
title: Presigned URL
description: Configuring lakeFS to use presigned URLs.
parent: Reference
nav_order: 100
has_children: false
---

# Presigned URL
{: .no_toc }

{% include toc.html %}

> lakeFS provides the option to access the data directly from the storage and not through lakeFS using presinged URL.

Reviewer: ```suggestion
With lakeFS, you can access data directly from the storage and not through lakeFS using a presined URL.
```

> lakeFS can generate a presigned URL to the object store by a user who has access to the object. The presigned URL can be used to access the object.

Reviewer: ```suggestion
Based on the user's access to an object in the object store, the presigned URL will get read or write access.
```

> lakeFS assume by default the object storage support presinged URLs, if not it should be disabled in the configuration.

Reviewer: The presign support is enabled for block adapter that supports it, and can be disabled by the configuration. Note that the UI support is disabled by default.

Author: done

# Using presigned URLs in the UI

> For the UI to fetch objects using a presigned URL to the object storage (instead through lakeFS), need to enable the presigned URL support UI in the lakeFS configuration and add CORS (Cross-Origin Resource Sharing) permissions to the bucket.

Reviewer: You must enable the presigned URL support UI (name the configuration used to enable it) in the lakeFS configuration (link the text to our configuration reference) and add CORS (Cross-Origin Resource Sharing) permissions to the bucket for the UI to fetch objects using a presigned URL (instead of through lakeFS).

Author: done
